src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -159,10 +166,34 @@ private Task<bool> NominateProjectAsync(string projectUniqueName, IVsProjectRest
             {
                 _logger.LogError(e.ToString());
                 TelemetryUtility.EmitException(nameof(VsSolutionRestoreService), nameof(NominateProjectAsync), e);
-                return Task.FromResult(false);
+                return false;
             }
         }
 
+        private async Task<ProjectNames> GetProjectNamesAsync(string projectUniqueName)
+        {
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            IVsSolution2 sln = ServiceLocator.GetGlobalService<SVsSolution, IVsSolution2>();
```

Comment:
When do we use IVsSolution2 vs IVsSolution5? How do you know which one to use?

Reply:
Whichever one has the API you want :)

Here, I need multiple pieces of information, so I get the `IVsHierarchy` object for the project, then call multiple other APIs to get the information from that. 

In the other class I used an API that let me get the project GUID from the project "unique name" (file path) in a single call, rather than getting the heirarchy object first, then get the GUID in another call. It was just luck that I found this other API.